app/shared/models/artifact.rb

```diff
@@ -6,27 +6,27 @@ class Artifact
     #
     # The Artifact model provides a simple interface to store a reference to any kind of data, but it is a dummy class.
     # It does not know how to access to its resource.
-    # Also, it holds a reference to the ArtifactProvider which created the Artifact itself.
+    # Also, it holds a reference to the ArtifactProvider which created the Artifact itself
     #
 
     # @return [UUID] The id of the artifact.
-    attr_accessor :id
+    attr_reader :id
 
     # type is just a String that refers to the kind of data that the Artifact represents (i.e., ipa, apk, log, etc.)
     # @return [String] The type or kind of artifact.
-    attr_accessor :type
+    attr_reader :type
 
     # @return [String] The reference to the artifact to a certain ArtifactProvider
-    attr_accessor :reference
+    attr_reader :reference
```

Comment:
Oh so this silently broke artifact support, see https://github.com/fastlane/ci/pull/508, as it's being accessed in sub-classes. Will this change also affect other parts of the code base, where we didn't consider sub classes accessing the attribute? In this case it's https://github.com/fastlane/ci/blob/master/app/shared/models/gcp_artifact_provider.rb#L76

cc @AndrewMcBurney @taquitos

Reply:
Yeah, we went through many rounds of testing and launching, but that doesn't mean we caught it all. I think the best course of action is to flex the system as much as we can.

Reply:
Yeah, it affects the provider one too 😳. I'll write a test suite for the artifact support so we won't run into problems like this is the future.